core/net/src/hybrid/service.rs

```diff
@@ -1116,6 +1121,38 @@ fn gen_id() -> u64 {
     rng.gen::<u64>() & 0x001f_ffff_ffff_ffff_u64
 }
 
+async fn bind_neighbourhood_bcast() -> anyhow::Result<()> {
+    let bcast_address = format!("{}/{}", net::local::BUS_ID, NewNeighbour::TOPIC);
+    bind_broadcast_with_caller(
+        &bcast_address,
+        move |_caller, _msg: SendBroadcastMessage<NewNeighbour>| async move {
+            if let Ok(client) = Net::client().await {
+                client.invalidate_neighbourhood_cache().await.ok();
+            }
+
+            return Ok(());
+        },
+    )
+    .await?;
+
+    Ok(())
```

Comment:
This async block already returns a `Result<(), BindBroadcastError>` so we just need to map the error into `anyhow::Error`.
```suggestion
    .await
    .map_err(Into::into)
```
Alternatively just return the `BindBroadcastError` and let the caller transform it if needed:
```
async fn bind_neighbourhood_bcast() -> Result<(), BindBroadcastError> {
...
    .await
}
````